### PR TITLE
Testing/mocking

### PR DESCRIPTION
**Pull request by npfrei**
Added DB tests for Activity DB and start of DataBaseManager

### CODE REVIEW

File: app/src/test/java/com/lastaoutdoor/lasta/data/db/ActivityRepositoryImplTest.kt
Comment (JoniDutoit) [inline app/src/test/java/com/lastaoutdoor/lasta/data/db/ActivityRepositoryImplTest.kt:1]: Great job on these tests! Don't hesitate to share the method on the notion that way we can reuse it later :)

File: app/src/test/java/com/lastaoutdoor/lasta/data/db/DatabasemanagerTest.kt
Comment (JoniDutoit) [inline app/src/test/java/com/lastaoutdoor/lasta/data/db/DatabasemanagerTest.kt:1]: Also great start on the Databasemanager. There are still overdue changes to this class but it's nice to have some tests for this class.

**Review by JoniDutoit — COMMENTED** [review]
Overall great job on the tests and thank you for finding a way of testing the database calls, it will be really useful in the future for the correctness of our algorithm and the line coverage.

**Review by JoniDutoit — APPROVED** [review]

**Review by Andy130604 — APPROVED** [review]
Really nice tests, maybe could implement the same way for different backend functions. This really is helping for the future

**Review by JeremyHugentobler — APPROVED** [review]
Thank you for your work !

This will help us a lot to test more complexe part of our code that were for now out of our reach.
Moreover your explanations on the notion are really helpful !!

**Pull request merged**